src/bin/help.ts

```diff
@@ -0,0 +1,172 @@
+import chalk from "chalk";
+
+import { allArgOptions } from "../shared/options/args.js";
+
+interface HelpTextSection {
+	sectionHeading: string;
+	subsections: {
+		flags: Flag[];
+		subheading: string | undefined;
+		warning: string | undefined;
+	}[];
+}
+
+interface Flag {
+	description: string;
+	flag: string;
+	type: string;
+}
+
+interface Option {
+	description: string;
+	docsSection:
+		| "core"
+		| "offline"
+		| "opt-out"
+		| "optional"
+		| "skip-disk"
+		| "skip-net";
+	multiple?: boolean;
+	type: string;
+}
+
+function logHelpTextSection(section: HelpTextSection): void {
+	console.log(" ");
+
+	console.log(`   ${chalk.black.bgGreenBright(section.sectionHeading)}`);
+
+	for (const subsection of section.subsections) {
+		if (subsection.warning) {
+			console.log(chalk.yellow(subsection.warning));
+		}
+
+		if (subsection.subheading) {
+			console.log(chalk.green(subsection.subheading));
+		}
+
+		for (const option of subsection.flags) {
+			const { description, flag, type } = option;
+			console.log(
+				chalk.cyan(
+					`
+      --${flag}${
+				type !== "boolean" ? ` (${chalk.cyanBright(type)})` : ""
+			}: ${description}`,
+				),
+			);
+		}
+	}
+}
+
+function createHelpTextSections(
```

Comment:
[Question] This function is only called once, is there a reason you wanted to pass an `options` parameter to it? E.g. I can see an argument for it being useful if it's tested directly as you'd suggested could be an option. If this file only ends up having a single snapshot test, though, I think there'd be a nice little code simplification of not having a param or needing the `Option` interface.

Reply:
I thought it would be easier to follow what was happening in the code if the `options` were pulled in in the exported function. I'm not set on this though, happy to change it if you think its better?

Reply:
I do :)

Reply:
Done!